### ISSUE TITLE
ORT 1.17.0 Release Candidates available for testing

### ISSUE THREAD

**Issue by YUNQIUGUO** [microsoft/onnxruntime#19236]
[ORT 1.17](https://github.com/microsoft/onnxruntime/wiki/Upcoming-Release-Roadmap) will be released in late January. Release candidate builds are available now for testing. If you encounter issues, please report them by responding in this issue.

Release branch: [rel-1.17.0](https://github.com/microsoft/onnxruntime/tree/rel-1.17.0)
Release manager: @YUNQIUGUO 

Pypi | Nuget | npm | Maven (Java)
--- | --- | --- | ---
CPU: [1.17.0.dev20240118001](https://aiinfra.visualstudio.com/PublicPackages/_artifacts/feed/ORT-Nightly/PyPI/ort-nightly/overview/1.17.0.dev20240118001)<br/>GPU: [1.17.0.dev20240118001](https://aiinfra.visualstudio.com/PublicPackages/_artifacts/feed/ORT-Nightly/PyPI/ort-nightly-gpu/overview/1.17.0.dev20240118001) | CPU: [1.17.0-dev-20240119-0139-a63b71eadb](https://aiinfra.visualstudio.com/PublicPackages/_artifacts/feed/ORT-Nightly/NuGet/Microsoft.ML.OnnxRuntime/overview/1.17.0-dev-20240119-0139-a63b71eadb)<br/>GPU (CUDA/TRT): [1.17.0-dev-20240118-2301-a63b71eadb](https://aiinfra.visualstudio.com/PublicPackages/_artifacts/feed/ORT-Nightly/NuGet/Microsoft.ML.OnnxRuntime.Gpu/overview/1.17.0-dev-20240118-2301-a63b71eadb)<br/>DirectML: [1.17.0-dev-20240119-0131-a63b71eadb](https://aiinfra.visualstudio.com/PublicPackages/_artifacts/feed/ORT-Nightly/NuGet/Microsoft.ML.OnnxRuntime.DirectML/overview/1.14.0-dev-20230123-0747-b51415b0ea)<br/>WindowsAI: [1.17.0-dev-20240119-0131-a63b71eadb](https://aiinfra.visualstudio.com/PublicPackages/_artifacts/feed/ORT-Nightly/NuGet/Microsoft.AI.MachineLearning/overview/1.17.0-dev-20240118-1027-a63b71e) | onnxruntime-node: [1.17.0-dev.20240118-a63b71eadb](https://www.npmjs.com/package/onnxruntime-node/v/1.17.0-dev.20240118-a63b71eadb)<br/>onnxruntime-react-native [1.17.0-dev.20240118-a63b71eadb](https://www.npmjs.com/package/onnxruntime-react-native/v/1.17.0-dev.20240118-a63b71eadb)<br/>onnxruntime-web [1.17.0-dev.20240118-a63b71eadb](https://www.npmjs.com/package/onnxruntime-web/v/1.17.0-dev.20240118-a63b71eadb)|CPU: [1.17.0-rc1](https://aiinfra.visualstudio.com/PublicPackages/_artifacts/feed/ORT-Nightly/maven/com.microsoft.onnxruntime%2Fonnxruntime/overview/1.17.0-rc1)<br/>GPU: [1.17.0-rc1](https://aiinfra.visualstudio.com/PublicPackages/_artifacts/feed/ORT-Nightly/maven/com.microsoft.onnxruntime%2Fonnxruntime_gpu/overview/1.17.0-rc1)

### Describe scenario use case

not applicable.

**Comment by fdwr** [comment]
@martinb35 / @smk2007

**Comment by tianleiwu** [comment]
Please share CUDA 12 packages for Python and Nuget.

**Comment by dbuades** [comment]
Agree, prebuilt CUDA 12.1 packages would really appreciated. Like you were doing with this nightly: https://aiinfra.visualstudio.com/PublicPackages/_artifacts/feed/ort-cuda-12-nightly/PyPI/ort-nightly-gpu/versions/1.17.0.dev20231205004

**Comment by YUNQIUGUO** [comment]
Here's the CUDA 12 version RC packages for Python and Nuget: (based on the same commit id)

 **ort-nightly-gpu** : [1.17.0.dev20240118002](https://aiinfra.visualstudio.com/PublicPackages/_artifacts/feed/ort-cuda-12-nightly/PyPI/ort-nightly-gpu/versions/1.17.0.dev20240118002)

**Microsoft.ML.OnnxRuntime.Managed:**[1.17.0-dev-20240118-2235-a63b71eadb](https://aiinfra.visualstudio.com/PublicPackages/_artifacts/feed/ort-cuda-12-nightly/NuGet/Microsoft.ML.OnnxRuntime.Managed/overview/1.17.0-dev-20240118-2235-a63b71eadb)

**Microsoft.ML.OnnxRuntime.Gpu:**  [1.17.0-dev-20240118-2235-a63b71eadb](https://aiinfra.visualstudio.com/PublicPackages/_artifacts/feed/ort-cuda-12-nightly/NuGet/Microsoft.ML.OnnxRuntime.Gpu/overview/1.17.0-dev-20240118-2235-a63b71eadb)

**Microsoft.ML.OnnxRuntime.Gpu.Linux:**[1.17.0-dev-20240118-2235-a63b71eadb](https://aiinfra.visualstudio.com/PublicPackages/_artifacts/feed/ort-cuda-12-nightly/NuGet/Microsoft.ML.OnnxRuntime.Gpu.Linux/overview/1.17.0-dev-20240118-2235-a63b71eadb)

**Microsoft.ML.OnnxRuntime.Gpu.Windows:**[1.17.0-dev-20240118-2235-a63b71eadb](https://aiinfra.visualstudio.com/PublicPackages/_artifacts/feed/ort-cuda-12-nightly/NuGet/Microsoft.ML.OnnxRuntime.Gpu.Windows/versions/1.17.0-dev-20240118-2235-a63b71eadb)

**Comment by sophies927** [comment]
@HectorSVC why was this issue unpinned? We pin our release candidates in GitHub issues so our partners + community members have easier access and can test them.

**Comment by YUNQIUGUO** [comment]
> Thanks for managing the release @YUNQIUGUO. Given a PyPI [release](https://pypi.org/project/onnxruntime/) has been made for 1.17, is there a plan to tag and release it on GitHub as well?

yep, we are still waiting for the last couple packages to be uploaded to the package management repo. after everything completed, will do a release announcement with the 1.17.0 package assets on Github as well.

**Comment by dbuades** [comment]
Thanks for the new release!
However, I see that a CUDA 12 version of the python onnxruntime-gpu package wasn't included in the pypy release. I tested the CUDA 12 RC packages that @YUNQIUGUO published in this thread last week and everything was working well, so I believe releasing a 1.17 version would be very useful.

Do you have plans to upload them at a later date or is there a particular reason why you are choosing not to do it?

Thank you very much!

**Comment by tianleiwu** [comment]
> Do you have plans to upload them at a later date or is there a particular reason why you are choosing not to do it?

@dbuades

You can install CUDA 12 python package like the following:
https://onnxruntime.ai/docs/install/#python-installs
```
pip install coloredlogs flatbuffers numpy packaging protobuf sympy
pip install onnxruntime-gpu --index-url https://pkgs.dev.azure.com/onnxruntime/onnxruntime/_packaging/onnxruntime-cuda-12/pypi/simple --no-cache-dir
```

**Comment by tianleiwu** [comment]
@YUNQIUGUO, please upload cuda 12 python package to https://aiinfra.visualstudio.com/PublicPackages/_artifacts/feed/onnxruntime-cuda-12/

**Comment by dbuades** [comment]
You are right @fvdnabee , thank you for reformulating my request.

**Comment by YUNQIUGUO** [comment]
@fvdnabee 

> The cuda 12 instructions are here: https://onnxruntime.ai/docs/install/#install-onnx-runtime-gpu-cuda-12x It seems the onnxruntime-gpu package doesn't exist on AzDo, I only found ort-nightly-gpu==1.17.0.dev20240130002 so far. Could we release a properly named artifact for 1.17, as opposed to the nightly build?

Thanks for reporting this. I am guessing our Python Cuda 12 packaging pipeline probably lacks a release version configuration/nightly build option. I actually uploaded `onnxruntime-gpu` to the Cuda 12 official feed but looks like it still contains a nightly in the naming though it should be an official one instead. I will contact the pipeline owner and we'll look into address this issue and re-upload an official named. 

Since it's the first time release cuda 12 version wheels to the feed, sorry about that - the issue hasn't be identified before.

@dbuades  I am not aware of a plan to upload it to an official repo like pypi.org yet. will ask around.

**Comment by 0x0480** [comment]
I'm curious if onnxruntime-node now supports dml and cuda?

**Comment by YUNQIUGUO** [comment]
@fs-eire

**Comment by fs-eire** [comment]
No. DML is ongoing (https://github.com/microsoft/onnxruntime/pull/19274) and CUDA support is the next.

> I'm curious if onnxruntime-node now supports dml and cuda?

**Comment by YUNQIUGUO** [comment]
@dbuades @fvdnabee  Hey, the issue has been resolved now. and here's the official onnxruntime-gpu package for cuda 12 version: https://aiinfra.visualstudio.com/PublicPackages/_artifacts/feed/onnxruntime-cuda-12/PyPI/onnxruntime-gpu/overview/1.17.0

**Comment by fdwr** [comment]
> I'm curious if there is any work towards DML for onnxruntime-java? Where might I be pointed to make that a thing? (I did search issues... seems not discussed yet?)

@CaelumF 🤔 I haven't heard desire that combination of language + backend before (more often DML is accessed via C++, Python, and C#), but is it already supported? I see this enum [ai.onnxruntime.OrtProvider.DIRECT_ML](https://javadoc.io/static/com.microsoft.onnxruntime/onnxruntime/1.17.0/ai/onnxruntime/OrtProvider.html#DIRECT_ML). (alas I don't know who on ORT owns the Java language layer to ask)

**Comment by CaelumF** [comment]
@fdwr hey thanks for responding. The pre built jars do have functions for enabling DirectML, but the binary wasn't compiled with DirectML enabled, and building with it enabled has some trouble. Issue posted here https://github.com/microsoft/onnxruntime/issues/19656 which also includes my use case in case you're curious

**Comment by Craigacp** [comment]
I'm the maintainer of the Java layer, I'll have a look at what's going on.
